### PR TITLE
[Docs]Private helm docs

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: docs/gitrepo-structure.md [hunk 1]
```diff
@@ -150,6 +150,21 @@ targetCustomizations:
...
+    
+    2. `cacerts` for custom CA bundle if helm repo is using custom CA.
+    
+    3. `ssh-privatekey` for ssh private key if repo is using ssh protocol.
+    
+    For example, to add a secret in kubectl, run 
+    
+    `kubectl create secret -n $namespace generic helm --from-literal=username=foo --from-literal=password=bar --from-file=cacerts=/path/to/cacerts --from-file=ssh-privatekey=/path/to/privatekey.pem`
+    
+    After secret is created, specify the secret to `gitRepo.spec.helmSecretName`. Make sure secret is created under the same namespace with gitrepo.
```
Comment (nickgerace) [inline docs/gitrepo-structure.md:166]: Is this supposed to be plaintext?
image

**Review by nickgerace — APPROVED** [review]
LGTM after seeing destination

File: docs/gitrepo-structure.md (hunk 1 above)
Comment (nickgerace) [reply docs/gitrepo-structure.md:166]: explained offline that this isn't pure MD. resolving

**Comment by bmdepesa** [comment]
Is it worth documenting that ssh keys with passphrase are not supported? I ran into this when testing.

**Comment by StrongMonkey** [comment]
@bmdepesa That's a good point. I will get this documented.

**Review by nickgerace — APPROVED** [review]
LGTM after requested changes above

File: docs/gitrepo-add.md [hunk 2]
```diff
@@ -49,6 +49,12 @@ spec:
   #
   # clientSecretName: my-ssh-key
   #
+  # If fleet.yaml contains private helm repo that requires autheticate,
```
Comment (catherineluse) [inline docs/gitrepo-add.md:52]: > If fleet.yaml contains a private Helm repo that requires authentication,
 > provide the credentials in a K8s secret and specify them here. Details are provided
 > in the fleet.yaml documentation.

**Review by catherineluse — CHANGES_REQUESTED** [review]

File: docs/gitrepo-structure.md [hunk 3]
```diff
@@ -150,6 +150,21 @@ targetCustomizations:
   clusterGroup: group1
 ```
 
+!!! hint "Private Helm Repo"
+    For private helm repo, user can reference a secret with the following key:
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:154]: For a private Helm repo,

File: docs/gitrepo-structure.md [hunk 4]
```diff
@@ -150,6 +150,21 @@ targetCustomizations:
...
+    
+    1. `username` and `password` for basic http auth if helm http repo is behind basic auth.
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:156]: if the Helm HTTP repo is behind basic auth.

File: docs/gitrepo-structure.md [hunk 5]
```diff
@@ -150,6 +150,21 @@ targetCustomizations:
...
+    
+    2. `cacerts` for custom CA bundle if helm repo is using custom CA.
```
Comment (catherineluse) [inline docs/gitrepo-structure.md:158]: if the Helm repo is using a custom CA.

**Comment by StrongMonkey** [comment]
@catherineluse Can you review this again? Comment have been addressed. Need it to be merged :)

**Review by catherineluse — APPROVED** [review]

**Pull request merged**